app/scripts/controllers/preferences.js

```diff
@@ -119,6 +119,18 @@ export default class PreferencesController {
     this.store.setMaxListeners(13);
     this.tokenListController = opts.tokenListController;
 
+    opts.onKeyringStateChange((state) => {
+      const accounts = new Set();
+      for (const keyring of state.keyrings) {
+        for (const address of keyring.accounts) {
+          accounts.add(address);
+        }
+      }
+      if (accounts.size > 0) {
+        this.syncAddresses(Array.from(accounts));
+      }
```

Comment:
Two things:

* Probably best to skip calling `syncAddresses` if the accounts array is empty, as you've done here, regardless of the issue with the return value. The wallet should never have zero addresses, except maybe when the vault is locked, and in that case we can wait until it's unlocked before syncing state. If we do want to start properly supporting an empty state, that would be something for the accounts team to take on later.
* Removing the `syncAddresses` return value makes sense to me! That doesn't seem useful.

Reply:
One thing to note is that upon unlock (`submitPassword`) `KeyringController` calls `updateIdentities`, which is bound to `PreferencesController.setAddresses` instead of `syncAddresses`. The only difference that I could spot is that `syncAddresses` does not change the account name (e.g. Account 1) if it is already present, while on the other hand `setAddresses` resets all names to `Account <i>`.

I don't think this will be an issue when we won't call `setAddresses` on unlock anymore, but something to keep in mind